[quant][be] Replace QConfigDynamic with QConfig in code

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #70009
* #69959
* #69879
* #69878
* #69877
* #69875
* __->__ #69864

Summary:
att, will have a follow up PR that removes QConfigDynamic in the api

Test Plan:
regression tests
```
python test/test_quantization.py TestPostTrainingStatic
python test/test_quantization.py TestPostTrainingDynamic
python test/test_quantization.py TestQuantizeFx
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D33073235](https://our.internmc.facebook.com/intern/diff/D33073235)

jerryzh168 committed:
[quant][be] Replace QConfigDynamic with QConfig in code

Summary:
att, will have a follow up PR that removes QConfigDynamic in the api

Test Plan:
regression tests
```
python test/test_quantization.py TestPostTrainingStatic
python test/test_quantization.py TestPostTrainingDynamic
python test/test_quantization.py TestQuantizeFx
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33073235).

jerryzh168 committed:
Update on "[quant][be] Replace QConfigDynamic with QConfig in code"


Summary:
att, will have a follow up PR that removes QConfigDynamic in the api

Test Plan:
regression tests
```
python test/test_quantization.py TestPostTrainingStatic
python test/test_quantization.py TestPostTrainingDynamic
python test/test_quantization.py TestQuantizeFx
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D33073235](https://our.internmc.facebook.com/intern/diff/D33073235)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33073235).

jerryzh168 committed:
Update on "[quant][be] Replace QConfigDynamic with QConfig in code"


Summary:
att, will have a follow up PR that removes QConfigDynamic in the api

Test Plan:
regression tests
```
python test/test_quantization.py TestPostTrainingStatic
python test/test_quantization.py TestPostTrainingDynamic
python test/test_quantization.py TestQuantizeFx
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D33073235](https://our.internmc.facebook.com/intern/diff/D33073235)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33073235).

jerryzh168 committed:
Update on "[quant][be] Replace QConfigDynamic with QConfig in code"


Summary:
att, will have a follow up PR that removes QConfigDynamic in the api

Test Plan:
regression tests
```
python test/test_quantization.py TestPostTrainingStatic
python test/test_quantization.py TestPostTrainingDynamic
python test/test_quantization.py TestQuantizeFx
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D33073235](https://our.internmc.facebook.com/intern/diff/D33073235)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33073235).

supriyar commented on torch/ao/quantization/qconfig.py:
```diff
@@ -80,64 +80,38 @@ def __new__(cls, activation, weight):
-default_dynamic_qconfig = QConfigDynamic(activation=default_dynamic_quant_observer,
-                                         weight=default_weight_observer)
+default_dynamic_qconfig = QConfig(activation=default_dynamic_quant_observer,
+                                  weight=default_weight_observer)
```
Should we have a warning first in case of OSS users who may still be using `QConfigDynamic` in their code? Deleting it would cause an error in the user code.

jerryzh168 replied:
do you mean we first put up a warning and then delete it in a future release?

jerryzh168 replied:
sure, will update the PR

jerryzh168 replied:
updated

jerryzh168 committed:
Update on "[quant][be] Replace QConfigDynamic with QConfig in code"


Summary:
att, will have a follow up PR that removes QConfigDynamic in the api

Test Plan:
regression tests
```
python test/test_quantization.py TestPostTrainingStatic
python test/test_quantization.py TestPostTrainingDynamic
python test/test_quantization.py TestQuantizeFx
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D33073235](https://our.internmc.facebook.com/intern/diff/D33073235)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][be] Replace QConfigDynamic with QConfig in code"


Summary:
att, will have a follow up PR that removes QConfigDynamic in the api

Test Plan:
regression tests
```
python test/test_quantization.py TestPostTrainingStatic
python test/test_quantization.py TestPostTrainingDynamic
python test/test_quantization.py TestQuantizeFx
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D33073235](https://our.internmc.facebook.com/intern/diff/D33073235)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][be] Replace QConfigDynamic with QConfig in code"


Summary:
att, will have a follow up PR that removes QConfigDynamic in the api

Test Plan:
regression tests
```
python test/test_quantization.py TestPostTrainingStatic
python test/test_quantization.py TestPostTrainingDynamic
python test/test_quantization.py TestQuantizeFx
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D33073235](https://our.internmc.facebook.com/intern/diff/D33073235)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33073235).

jerryzh168 committed:
Update on "[quant][be] Replace QConfigDynamic with QConfig in code"


Summary:
att, will have a follow up PR that removes QConfigDynamic in the api

Test Plan:
regression tests
```
python test/test_quantization.py TestPostTrainingStatic
python test/test_quantization.py TestPostTrainingDynamic
python test/test_quantization.py TestQuantizeFx
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D33073235](https://our.internmc.facebook.com/intern/diff/D33073235)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33073235).

jerryzh168 committed:
Update on "[quant][be] Replace QConfigDynamic with QConfig in code"


Summary:
att, will have a follow up PR that removes QConfigDynamic in the api

Test Plan:
regression tests
```
python test/test_quantization.py TestPostTrainingStatic
python test/test_quantization.py TestPostTrainingDynamic
python test/test_quantization.py TestQuantizeFx
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D33073235](https://our.internmc.facebook.com/intern/diff/D33073235)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33073235).
